lib/bolt/puppetdb/client.rb

```diff
@@ -41,22 +45,35 @@ def facts_for_node(certnames)
 
       def make_query(query, path = nil)
         body = JSON.generate(query: query)
-        url = "#{@config.uri}/pdb/query/v4"
+        url = "#{uri}/pdb/query/v4"
         url += "/#{path}" if path
 
         begin
           response = http_client.post(url, body: body, header: headers)
+        rescue SocketError, OpenSSL::SSL::SSLError, SystemCallError, Net::ProtocolError, IOError => err
```

Comment:
These appear to be the only errors that pdb fails over for. I'm not sure why we wouldn't just fail over on any StandardError though.

Reply:
The only reason I have for not failing over and StandardError is if there are a lot of server urls, and the issue is going to cause them all to fail, failing fast rather than having to wait and try each. That doesn't seem like it will be common though, and I think it'll be more annoying to expect failover for any errors and have that not happen.